src/hooks/useBalance.tsx

```diff
@@ -0,0 +1,19 @@
+import { useGlobalState, useWalletState } from '@src/state';
+import { useQuery } from '@tanstack/react-query';
+
+export function useBalance(address?: string) {
+  const [{ arioContract }] = useGlobalState();
+  const [{ walletAddress }] = useWalletState();
+
+  return useQuery({
+    queryKey: [walletAddress?.toString(), arioContract],
+    queryFn: async () => {
+      if (!address && !walletAddress)
+        throw new Error('No address provided to fetch balance');
+      const arioBalance = await arioContract.getBalance({
+        address: (address ?? walletAddress?.toString()) as string,
```

Comment:
Does this need nullish check on address  if it's checking !address earlier?

Reply:
Yup because its only exiting if both address AND walletAddress are nil, so we need to have this check